doc/languages-frameworks/julia.section.md

```diff
@@ -0,0 +1,23 @@
+# Julia {#sec-julia}
+
+### Installing Julia Packages {#sssec-installing-julia-packages}
+
+Most Julia compilers available in Nixpkgs have a passthru attribute
+called `withPackages`. This function permits creating a collection of
+packages ready for use in Julia as in many other language frameworks.
+For example, to make the package `Plots` available to Julia you can run
+
+```ShellSession
+$ nix-shell -p 'julia-bin.withPackages (p: with p; [ Plots ])'
+```
+
+The list of Julia packages in Nixpkgs is accessible through the
+compiler passthru attribute set `juliaPackages`, for example
+`julia-bin.juliaPackages`.
+
+Note that, if you try to add a nixpkgs provided package to a Julia
+environment when a newer version is available upstream, the Julia
+package manager `Pkg` will prefer the latter. To prevent this you can
```

Comment:
So the environments are completely impure and you will always get the latest?
This would completely unacceptable for nixpkgs.

Reply:
No. When you add a package through `withPackages` you can just `import` it in Julia and you'll get the nixpkgs version. However, if you need a package not in nixpkgs then you typically create a Julia environment.
 Julia environments are similar to Phython virtual environments and you create them with the help of the Julia package manager `Pkg`. By default when you add a package it looks for the newest version even if there is a version available on your system. That's a Julia developers decision, not a nixpkgs one. If you don't want that you can do as I explained. A user can simply set an environment variable to always work offline if he so wants.
After the environment is created all package versions are fixed.